Configuration/Applications/python/ConfigBuilder.py

```diff
@@ -1676,6 +1677,17 @@ def prepare_PAT(self, sequence = "miniAOD"):
 
         return
 
+    def prepare_PATGEN(self, sequence = "miniAOD"):
+        ''' Enrich the schedule with PATGEN '''
+        self.loadDefaultOrSpecifiedCFF(sequence,self.PATDefaultCFF,1) #this is unscheduled
```

Comment:
mmm, I am not sure I understand sorry.
   * I am not sure how I can remove the PATGEN and put it back conditionally into PAT, because PAT can run both on data and MC. Which would be the condition?
   * I am not sure what to add similarly to `self.labelsToAssociate.append('patTask')` because if I look at the patTask definition [*] I don't see anything relevant for the MINIGEN
   * I was indeed searching for a cmsDriver and you provided it thanks, I modified it like this [**] and fixed the trivial bug

this is getting long and complicated, thanks for the patience and sorry for the mess......



[*] https://github.com/cms-sw/cmssw/blob/3f02fc46bb777969fcd88ab7d707241050353762/Configuration/StandardSequences/python/PAT_cff.py

[**] `cmsDriver.py step1 --mc  --conditions auto:phase1_2017_realistic --era Run2_2017 --eventcontent MINIGEN -s PATGEN --runUnscheduled --datatier MINIGEN  -n -1 --filein /store/mc/RunIISummer15GS/DYToMuMu_M_50_TuneAZ_8TeV_pythia8/GEN-SIM/GenOnly_MCRUN2_71_V1-v3/100000/006A257F-D140-E711-93E2-0025901C1A92.root --fileout file:step2.root --no_exec -n 1000`